test/e2e/operator_test.go

```
@@ -2139,6 +2139,290 @@ func TestContainerLogging(t *testing.T) {
+	}
+	if len(podList.Items) != 1 {
+		var podNames []string
+		for i := range podList.Items {
+			podNames = append(podNames, podList.Items[i].Name)
+		}
+		t.Fatalf("expected ingress controller %s to have exactly 1 router pod, but it has %d: %s", ic.Name, len(podList.Items), strings.Join(podNames, ", "))
+	}
+
+	// Make a request to the console route.
+	routeName := types.NamespacedName{Namespace: "openshift-console", Name: "console"}
+	route := &routev1.Route{}
+	if err := kclient.Get(context.TODO(), routeName, route); err != nil {
+		t.Fatalf("failed to get the console route: %v", err)
+	}
+	clientPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "headertest",
+			Namespace: podList.Items[0].Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "curl",
+					Image:   podList.Items[0].Spec.Containers[0].Image,
+					Command: []string{"/bin/curl"},
+					Args: []string{
+						"-k",
+						"-o", "/dev/null", "-s",
+						"--resolve",
+						route.Spec.Host + ":443:" + podList.Items[0].Status.PodIP,
+						"https://" + route.Spec.Host,
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+	if err := kclient.Create(context.TODO(), clientPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
+			if apierrors.IsNotFound(err) {
+				return
+			}
+			t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+		}
+	}()
+
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %v", err)
+	}
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
+
+	err = wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
+		for _, pod := range podList.Items {
```

Same comment as above to remove the `for` loop and return the error instead of using `continue`.